pints/_error_measures.py

```diff
@@ -214,21 +214,32 @@ class MeanSquaredError(ProblemErrorMeasure):
     ``problem``
         A :class:`pints.SingleOutputProblem` or
         :class:`pints.MultiOutputProblem`.
+    ``weights``
+        (Optional) A sequence of (float) weights, exactly one per problem
+        output. If no weights are specified all sums will be weighted equally.
     """
-    def __init__(self, problem):
+    def __init__(self, problem, weights=None):
         super(MeanSquaredError, self).__init__(problem)
         self._ninv = 1.0 / np.product(self._values.shape)
 
+        if weights is None:
+            weights = [1] * self._n_outputs
+        elif self._n_outputs != len(weights):
+            raise ValueError(
+                'Number of weights must match number of problem outputs.')
+        # Check weights
+        self._weights = np.asarray([float(w) for w in weights])
+
     def __call__(self, x):
-        return (np.sum((self._problem.evaluate(x) - self._values)**2) *
-                self._ninv)
+        return (np.sum((self._problem.evaluate(x) - self._values)**2 *
```

Comment:
Could you rewrite this with 2 sums to make it a bit faster?

The procedure should become:

1. Evaluate and subtract values (2d)
2. Sum along the time axis
3. Multiply by weights
4. Sum along the output axis
5. Multiply by n_inv and return

Reply:
Actually, could you do some benchmarking and see if this is necessary?
If so, we'll need to do it below as well

Reply:
Just did a little test, it is necessary, so I will apply this change to everywhere whenever possible.